proposals/0000-app-setup.md

```diff
@@ -0,0 +1,184 @@
+- Start Date: 2022-10-03
+- Reference Issues: <!-- related issues, otherwise leave empty -->
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+This RFC proposes a unified interface that provides a hook into island **setup**. User-provided hooks should run for both server rendering and client-side hydration.
```

Comment:
Can you define what setup is?

Reply:
Rephrased this! Hopefully it is more clear now.